### DISCUSSION TITLE
How to run Android Studio Emulator via WSL2

### DISCUSSION THREAD

**Discussion by DiegoMcDipster** [microsoft/WSL#6625] [Q&A]
Hi,

Does anyone know where I can find the step by step instructions to running Android Studio Emulator via WSL2?

What I did was:
0. I'm running Node version: 14.15.4 and expo-cli version: 4.2.1
1. installed Android studio for Windows (https://developer.android.com/studio)
2. Setup an Android Virtual Device and ran it
3. Updated by zshrc file with:
export ANDROID_SDK=/mnt/c/Users/user/AppData/Local/Android/Sdk
export PATH=/mnt/c/Users/user/AppData/Local/Android/Sdk/platform-tools:$PATH
* on the SDK manager my path is: C:\Users\user\AppData\Local\Android\Sdk
4. Created an Expo app (blank managed workflow) in WSL2.  Edited the text via Visual Studio code
5. to start the Expo server I ran: npm start  
6. to run with android emulator pressed: a

This gives me the error:
Couldn't start project on Android: No Android connected device found, and no emulators could be started automatically.
Please connect a device or create an emulator (https://docs.expo.io/workflow/android-studio-emulator).

The docs on expo cover how to setup on windows or ios or linux.  But not on WSL2.

Note: running the emulator in the web browser does work

Your help and advice would be really appreciated.

**Comment by Biswa96** [comment]
Many has achieved it here https://github.com/microsoft/WSL/issues/6471

**Reply by DiegoMcDipster** [reply]
Thanks for taking the time to reply.

However, that's not a step by step guide.  Also it's an issue with a windows build where people had it running before and it stopped running.

I'm still at the stage of setting it up.

**Comment by kenneth0417** [comment]
Hi, any updates for this?

**Comment by Dozr13** [comment]
I'm also curious as I feel I've tried everything I can think of.. Basically my exact issue, so any updates would be awesome!

**Comment by DiegoMcDipster** [comment]
Man o man, this was something I raised over two years ago!

In the end I was distracted by work stuff and never got round to doing that course that i needed the emulator for... so you could say that i just gave up.

So, unfortunately, this is still an open issue.

**Reply by Dozr13** [reply]
I feel ya man, I saw the dates haha but dang it, I'm so close right now, I got the emulators up and working through WSL2 and was hoping there could be someone out there that knew my missing piece or at least a direction to point me.

Thanks for the reply, and glad you got out of a spot of worrying about it. haha

**Reply by CosmicTiger** [reply]
> I feel ya man, I saw the dates haha but dang it, I'm so close right now, I got the emulators up and working through WSL2 and was hoping there could be someone out there that knew my missing piece or at least a direction to point me.
> 
> Thanks for the reply, and glad you got out of a spot of worrying about it. haha

Is there a way you could explain to me further how you did that? @Dozr13

**Comment by marins1897** [comment]
Create directories up to the Windows path:
`mkdir -p /mnt/c/Users/<YOUR_USERNAME>/AppData/Local/Android/Sdk/platform-tools`

Create a symlink to working ADB:
`ln -s $(which adb) /mnt/c/Users/<YOUR_USERNAME>/AppData/Local/Android/Sdk/platform-tools/adb`

This creates the fake adb binary that points to real Linux adb that works in WSL.

Run the app.
